lib/process-services/process-list/components/start-process.component.ts

```diff
@@ -89,22 +92,43 @@ export class StartProcessInstanceComponent implements OnChanges {
             this.moveNodeFromCStoPS();
         }
 
-        let appIdChange = changes['appId'];
-        let appId = appIdChange ? appIdChange.currentValue : null;
-        this.load(appId);
+        if (changes['appId'] && changes['appId'].currentValue) {
+            this.appId = changes['appId'].currentValue;
+        }
+
+        this.loadStartProcess();
     }
 
-    public load(appId?: number) {
+    public loadStartProcess() {
         this.resetSelectedProcessDefinition();
         this.resetErrorMessage();
-        this.activitiProcess.getProcessDefinitions(appId).subscribe(
-            (res) => {
-                this.processDefinitions = res;
-            },
-            () => {
-                this.errorMessageId = 'ADF_PROCESS_LIST.START_PROCESS.ERROR.LOAD_PROCESS_DEFS';
-            }
-        );
+
+        if (this.appId) {
+            this.activitiProcess.getProcessDefinitions(this.appId).subscribe(
+                (processDefinitionRepresentations: ProcessDefinitionRepresentation[]) => {
+                    this.processDefinitions = processDefinitionRepresentations;
+
+                    console.log('this.processDefinitions.length ' + this.processDefinitions.length);
+
+                    if (this.processDefinitions.length === 1) {
+                        this.currentProcessDef = JSON.parse(JSON.stringify(this.processDefinitions[0]));
+                    } else {
+                        if (this.processDefinitionId) {
+                            this.processDefinitions = this.processDefinitions.filter((currentProcessDefinition) => {
```

Comment:
what happens if the defined processDefinitionId is not part of the processDefinitions list?

Reply:
is not going to show any process because of the wrong parameter, is the same also if you add a wrong aappid